### ISSUE TITLE
Attempt to play a Widevidne DRM stream results in Kodi crash and exit/restart

### ISSUE THREAD

**Issue by cralin** [peak3d/inputstream.adaptive#1103]
### Describe the bug
Here is a clear and concise description of what the problem is:

Attempt to play a Widevidne DRM stream results in Kodi crash and exit.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Stream is played successfully. 
Kodi does not crash and exit

## Actual Behavior

Stream does not play, Kodi crashes and exit

### To Reproduce

1. Install Kodi 20 (Nexus) from play store
2. Install add-on using imputstream.adaptive to play Widevine stream 
3. Attempt to play Widevine DRM protected stream

### Debuglog

The debuglog can be found here:
[kodi.log](https://github.com/xbmc/inputstream.adaptive/files/10473492/kodi.log)

### MPD/M3U8s/ISM

An example or copy of a manifest (or manifests for HLS - master and variants) can be found here:
[manifest.mpd.txt](https://github.com/xbmc/inputstream.adaptive/files/10473497/manifest.mpd.txt)

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

Same addon running on Kodi 19 (Matrix) on the same Android and RPI4 hardware plays just fine the same stream.

### Your Environment
Used Operating system:

 - [X] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Android ARM 32-bit, API level 21
 - Kodi version: 20.0 (Nexus)
 - inputstream.adaptive version: 20.3.2

*note: Once the issue is made we require you to update it with new information should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by cralin** [comment]
It looks like the same behavior on a RPI4 with 2GB RAM running LibreELEC (community): nightly-20230120-f3ef152 and inputstream.adaptive v20.3.2.1

[kodi_crashlog_20230122030555.log](https://github.com/xbmc/inputstream.adaptive/files/10473588/kodi_crashlog_20230122030555.log)

**Comment by glennguy** [comment]
Hi @cralin 
Would you be able to supply what you're setting for the license_key property? Suspecting it might be similar/same as https://github.com/xbmc/inputstream.adaptive/issues/1098

I've loaded the manifest locally and given a dummy license_key and it gets further than your logs do

**Comment by cralin** [comment]
@glennguy 

In the add-on code I do this:

```
      is_helper = inputstreamhelper.Helper('mpd', drm='com.widevine.alpha')
      if is_helper.check_inputstream():
        play_item = xbmcgui.ListItem(path=_stream_details_['stream']['abr'])
        play_item.setProperty('inputstream', 'inputstream.adaptive')
        play_item.setProperty('inputstream.adaptive.stream_headers', _headers_)
        play_item.setProperty('inputstream.adaptive.license_type', 'com.widevine.alpha')
        play_item.setProperty('inputstream.adaptive.manifest_type', 'mpd')
        play_item.setProperty('inputstream.adaptive.license_key', _stream_details_['stream']['proxy'] + '|' + _lic_headers_ + '|R{SSM}|')
        play_item.setMimeType('application/dash+xml')

        # Pass the item to the Kodi player.
        xbmcplugin.setResolvedUrl(int(common_vars.__handle__), True, listitem=play_item)


```
Where:

`_headers_ = &User-Agent=Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0`

`_lic_headers_ = &User-Agent=Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0&Connection=keep-alive&Accept-Encoding=identity&verifypeer=false`

`_stream_details_['stream']['abr'] = https://edge81.rcs-rds.ro/V5-9154162-7b4e5c4b81319deaf578e1470408cd0d-1450744194-0-1674303099-4-digi24-033aafb038f62947/digionline/digi24.stream/manifest.mpd?is=4&ns=digi24&p=Android&s=app&sn=ro.rcsrds.digionline&t=00000000000000000000000000000000&t4=V4-2754077-90ecfa48e47cd638a4e4301494fe84cc-1450744194-0-1674303099-4-digi24&t5=V5-9154162-7b4e5c4b81319deaf578e1470408cd0d-1450744194-0-1674303099-4-digi24-033aafb038f62947`

`_stream_details_['stream']['proxy'] = https://edge81.rcs-rds.ro/V5-9154162-7b4e5c4b81319deaf578e1470408cd0d-1450744194-0-1674303099-4-digi24-033aafb038f62947/drm/digionline`

**Comment by dimkroon** [comment]
Yes, as @glennguy mentioned it's very likely the same as #1098, but now with a leading ampersand instead of trailing.

I tried the test addon again from #1098 and moved the ampersand from the end to the start of the license request headers. On Nexus/IA 20.3.2 it crashes, but with the IA test build provided in #1098 it runs OK (on win10).

So it's most likely already fixed in #1099 
But you can also just remove the leading '&' in `_lic_headers_`, and also in `_headers_` for that matter, since it doesn't really make sense.

**Comment by cralin** [comment]
@glennguy , @dimkroon,

I can confirm that if I remove the leading '&' in `_lic_headers_` and in `_headers_` the stream plays as expected and Kodi no longer crashes and restarts.

Tested on both RPI4 and on Android TV devices mentioned in my previous comments
